spec/features/trainee_actions/undo_withdraw_trainee_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Undo trainee withdrawal" do
+  include SummaryHelper
+
+  before do
+    ActiveJob::Base.queue_adapter.enqueued_jobs.clear
+    ActiveJob::Base.queue_adapter.perform_enqueued_jobs = false
+    when_i_am_on_the_undo_withdrawal_page
+  end
+
+  after do
+    ActiveJob::Base.queue_adapter.perform_enqueued_jobs = true
+  end
+
+  context "validations" do
+    context "comment validation" do
+      scenario "no comment provided" do
+        click_button "Continue"
+        expect(page).to have_content("Enter a comment")
+      end
+    end
+
+    context "ticket validation" do
+      scenario "invalid URL" do
+        fill_in "undo-withdrawal-form-comment-field", with: "asdf"
+        fill_in "undo-withdrawal-form-ticket-field", with: "asdf"
+        click_button "Continue"
+        expect(page).to have_content("Must be a URL")
+      end
+    end
+  end
+
+  context "undo trainee withdrawal flow" do
+    scenario "successfully undo trainee withdrawal" do
+      fill_in "undo-withdrawal-form-comment-field", with: "this is the comment"
```

Comment:
Can we wrap some of the low level steps into sentences? I know it's a bit of a pain, but we should try and be consistent with the other feature files.